tests/Microsoft.Identity.Test.Unit/PublicApiTests/OBOTests.cs

```diff
@@ -791,6 +792,53 @@ public async Task AcquireTokenByObo_NormalOboThenLongRunningInitiate_WithTheSame
             }
         }
 
+        [TestMethod]
+        public async Task AcquireTokenByObo_LongRunningThenRemoveTokens_TestAsync()
```

Comment:
You'll need some test with custom caching to see how the TokenCacheNotification args behave.